utils/TaskPeriodUtils.js

```diff
@@ -0,0 +1,70 @@
+export const months = [
+  "December",
+  "January",
+  "February",
+  "March",
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+];
+
+export function getIntervals(months) {
+  // Function to get cartesian product of multiple arrays
+  // See https://stackoverflow.com/questions/4331092/finding-all-combinations-cartesian-product-of-javascript-array-values
+  function* cartesian(head, ...tail) {
+    let remainder = tail.length ? cartesian(...tail) : [[]];
+    for (let r of remainder) for (let h of head) yield [h, ...r];
+  }
+
+  // Get cartesian of months and within-months periods
+  const intervals = [...cartesian(["early", "mid", "late"], months)].map(
+    (key) => key.join(" ")
+  );
+
+  return intervals;
+}
+
+export function getCurrentInterval(months) {
+  const date = new Date();
+  const month = months[date.getMonth() + 1];
+  const day = date.getDate();
+  const currentInterval =
+    (day < 11 ? "early " : day < 21 ? "mid " : "late ") + month;
+
+  return currentInterval;
+}
+
+export function getActiveTasksByPlant(plants, months) {
```

Comment:
I'm just wondering a little bit: You export the `months` array just to get it back as a parameter?